spec/models/journal_notification_mailer_spec.rb

```diff
@@ -0,0 +1,156 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+require 'spec_helper'
+
+describe JournalNotificationMailer do
+  let(:project) { FactoryGirl.create(:project_with_types) }
+  let(:user) do
+    FactoryGirl.build(:user,
+                      mail_notification: 'all',
+                      member_in_project: project)
+  end
+  let(:work_package) {
+    FactoryGirl.create(:work_package,
+                       project: project,
+                       author: user,
+                       type: project.types.first)
+  }
+  let(:notifications) { [] }
+
+  before do
+    allow(User).to receive(:current).and_return(user)
+    allow(Setting).to receive(:notified_events).and_return(notifications)
+
+    ActionMailer::Base.deliveries.clear
+  end
+
+  shared_examples_for 'handles deliveries' do |notification_setting|
+    context 'setting enabled' do
+      let(:notifications) { [notification_setting] }
+
+      it 'sends a notification' do
+        expect(ActionMailer::Base.deliveries.size).to eq(1)
+      end
+
+      context 'insufficient work package changes' do
+        let(:another_work_package) {
+          FactoryGirl.create(:work_package,
+                             project: project,
+                             author: user,
+                             type: project.types.first)
+        }
+        before do
+          another_work_package.add_journal(user)
+          another_work_package.description = 'needs more changes'
```

Comment:
I am not sure if I miss something here, but this will work for ``work_package_note_added`` because the work package under test does not have any notes attached. Hence ``journal.notes.present?`` is false which in turn prevents a notification to be sent.

Is it possible that you confused description with notes?